packages/runtime-core/src/main/kotlin/elide/runtime/core/PolyglotEngineConfiguration.kt

```diff
@@ -31,6 +33,77 @@ import elide.runtime.core.PolyglotEngineConfiguration.HostAccess.ALLOW_NONE
     ALLOW_NONE,
   }
 
+  /**
+   * Provides read-only information about the Host platform, which can be used by plugins to run platform-specific
+   * code.
+   */
+  public data class HostPlatform(
```

Comment:
### Host Platform

The `HostPlatform` class and its companion provide functions to resolve host information, and to parse individual components from generic sources (mainly `os.name` and `os.arch` JVM System Properties).

Two enums, `OperatingSystem` and `Architecture`, provide well-known values for OS families and architectures, which can then be combined in a single `HostPlatform` record.